src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -128,29 +176,64 @@ def check_align_starting_time_internal(self):
         expected_timestamps = unaligned_timestamps + starting_time
         assert_array_equal(x=aligned_timestamps, y=expected_timestamps)
 
-    def check_align_timestamps_internal(self):
-        """Ensure that internal mechanisms for the timestamps getter/setter work as expected."""
-        unaligned_timestamps = self.interface.get_original_timestamps()
+    def check_interface_align_timestamps_mutation(self):
+        """Check aligning the timestamps for the interface does not change the value of .get_original_timestamps()."""
+        fresh_interface = self.data_interface_cls(**self.test_kwargs)
+        pre_alignment_original_timestamps = fresh_interface.get_original_timestamps()
 
-        aligned_timestamps = unaligned_timestamps + 1.23 + np.random.random(size=unaligned_timestamps.shape)
-        self.interface.align_timestamps(aligned_timestamps=aligned_timestamps)
+        aligned_timestamps = pre_alignment_original_timestamps + 1.23
+        fresh_interface.align_timestamps(aligned_timestamps=aligned_timestamps)
 
-        retrieved_aligned_timestamps = self.interface.get_timestamps()
-        assert_array_equal(x=retrieved_aligned_timestamps, y=aligned_timestamps)
+        post_alignment_original_timestamps = fresh_interface.get_original_timestamps()
+        assert_array_equal(x=post_alignment_original_timestamps, y=pre_alignment_original_timestamps)
 
-    def check_align_starting_time_external(self):
-        pass  # TODO: generalize
+    def check_interface_interpolation(self):
```

Comment:
I don't think this should be here. 

This right now is testing that the tested interface can be used as a source of timestamps for a third interface that you create here on the fly (i.e. `MockBehaviorEventInterface`).  Given that all the other methods used before are already tested somewhere else this seems redundant. 

The interface that you want to test `MockBehaviorEventInterface` is inicidental to this. This should be a test of its own I feel and does not need to be in the general mechanism just yet.

Reply:
Oh I see; the point was for this to test the interpolation method for this interface, but will have to think about what that would mean more generally

Reply:
which is hard to do for arbitrary interfaces now that I think about it... especially higher frequency recordings, or format specific timestamps

Just removing it in that case

Reply:
Removed